src/download.tsx

```diff
@@ -158,7 +165,10 @@ class Download extends KalturaPlayer.core.BasePlugin {
 
   reset() {
     this.upperBarManager?.remove(this.iconId);
+    // @ts-ignore
+    this.player.getService('AudioPluginsManager').remove(this.audioPlayerIconId)
```

Comment:
you can add a getter for the service

Reply:
too verbose - unnecessarily - no real benefits